test/choosetests.jl

```diff
@@ -82,60 +82,29 @@ function choosetests(choices = [])
         tests = testnames
     end
 
-
-    unicodetests = ["unicode/utf8"]
-    if "unicode" in skip_tests
-        filter!(x -> (x != "unicode" && !(x in unicodetests)), tests)
-    elseif "unicode" in tests
-        # specifically selected case
-        filter!(x -> x != "unicode", tests)
-        prepend!(tests, unicodetests)
-    end
-
-    stringtests = ["strings/basic", "strings/search", "strings/util",
-                   "strings/io", "strings/types"]
-    if "strings" in skip_tests
-        filter!(x -> (x != "strings" && !(x in stringtests)), tests)
-    elseif "strings" in tests
-        # specifically selected case
-        filter!(x -> x != "strings", tests)
-        prepend!(tests, stringtests)
-    end
-
-    # do subarray before sparse but after linalg
-    if "subarray" in skip_tests
-        filter!(x -> x != "subarray", tests)
-    elseif "subarray" in tests
-        filter!(x -> x != "subarray", tests)
-        prepend!(tests, ["subarray"])
-    end
-
-    compilertests = ["compiler/inference", "compiler/validation", "compiler/ssair", "compiler/irpasses",
-                     "compiler/codegen", "compiler/inline", "compiler/contextual"]
-
-    if "compiler" in skip_tests
-        filter!(x -> (x != "compiler" && !(x in compilertests)), tests)
-    elseif "compiler" in tests
-        # specifically selected case
-        filter!(x -> x != "compiler", tests)
-        prepend!(tests, compilertests)
-    end
-
-    if "stdlib" in skip_tests
-        filter!(x -> (x != "stdlib" && !(x in STDLIBS)) , tests)
-    elseif "stdlib" in tests
-        filter!(x -> (x != "stdlib" && !(x in STDLIBS)) , tests)
-        prepend!(tests, STDLIBS)
-    end
-
-    # do ambiguous first to avoid failing if ambiguities are introduced by other tests
-    if "ambiguous" in skip_tests
-        filter!(x -> x != "ambiguous", tests)
-    elseif "ambiguous" in tests
-        filter!(x -> x != "ambiguous", tests)
-        prepend!(tests, ["ambiguous"])
-    end
-
+    function patchlib!(skip_tests,tests,name,files=[name]; stdlib::Bool=false)
+       flt = x -> (x != name && !(x in files))
+       testsflt = stdlib ? flt : !=(name)
```

Comment:
I would believe you can just delete `testsflt`, and that using (the equivalent of) `!=(name)` in most code above was an oversight, as it would be unexpected from the user to specify both the `name` and the `files`. Otherwise, some file can end up appearing twice in `tests` and I don't think it can be useful. Would be good if someone can confirm (maybe @vtjnash?)

Reply:
I have not change so much code there. May be shall we wait for a confirmation ?

Reply:
I suggest that you update your PR according to my comment (i.e. use the same `flt` function in both `filter!` calls, and remove the `stdlib` keyword), and then I will merge. I can't think of a good reason to keep this distinction, and I will take responsability for this if anyone complains :)